calcom/cal.com
# #4306: Added darkmode support for select by styling via tailwind

## What does this PR do?

This PR implements **dark mode for all selects** (had to rebuild to tailwind to properly use dark mode styles), and converted the **timezoneselect component to use the new select v2**, so this also looks nice and supports dark mode (part of #4508).
Also added a sexy custom scroll bar (css only) to this dropdown, just looks so much better 😉 

Fixes #4306 #4508

<img width="1039" alt="CleanShot 2022-10-04 at 15 53 33@2x" src="https://user-images.githubusercontent.com/2969573/193838013-8b67b70a-28b6-4849-adff-7ed86e33370e.png">

## Loom video

In the below loom video I'll show you the different variants of the select that are still around, including the new dark mode in the beginning of the video. At the end of the video I also show a local testpage with a multiselect variant that has also been styled to match dark mode.

https://www.loom.com/share/bad49b3209264ccdb2dce864ebdd4b7a

**Environment**: Staging(main branch)

## Type of change



## How should this be tested?

This changed touched all v2 selects. It is very important to click through the platform and double check we didn't break any of them.

- [ ] Ensure we didn't break any selects by consolidating them to one.
